mm2src/coins/coin_errors.rs

```diff
@@ -0,0 +1,442 @@
+use derive_more::Display;
+use hex::FromHexError;
+use mm2_number::bigdecimal::ParseBigDecimalError;
+use spv_validation::helpers_validation::SPVError;
+
+use crate::{eth::{TokenDecimalsError, TryToAddressError, Web3RpcError},
+            qrc20::script_pubkey::ScriptExtractionError,
+            utxo::{qtum::ScriptHashTypeNotSupported, rpc_clients::UtxoRpcError, utxo_builder::UtxoConfError,
+                   BroadcastTxErr, UnsupportedAddr},
+            NumConversError, UnexpectedDerivationMethod};
+
+#[derive(Debug, Display)]
+pub enum CheckPaymentSentError {
+    AddressParseError(String),
+    AddrImportFailed(String),
+    Erc20PaymentDetailsError(String),
+    DeserializationErr(String),
+    JsonRpcError(JsonRpcError),
+    PaymentStatusError(String),
+    PublicKeyErr(String),
+    SignTxError(String),
+    TransportError(String),
+    TryToAddressError(TryToAddressError),
+    UtxoRpcError(UtxoRpcError),
+    UnexpectedDerivationMethod(UnexpectedDerivationMethod),
+}
+
+impl From<keys::Error> for CheckPaymentSentError {
+    fn from(err: keys::Error) -> Self { Self::PublicKeyErr(err.to_string()) }
+}
+
+impl From<PaymentStatusError> for CheckPaymentSentError {
+    fn from(err: PaymentStatusError) -> Self { Self::PaymentStatusError(err.to_string()) }
+}
+
+impl From<JsonRpcError> for CheckPaymentSentError {
+    fn from(err: JsonRpcError) -> Self { Self::JsonRpcError(err) }
+}
+
+impl From<UtxoRpcError> for CheckPaymentSentError {
+    fn from(err: UtxoRpcError) -> Self { Self::UtxoRpcError(err) }
+}
+
+impl From<AddressParseError> for CheckPaymentSentError {
+    fn from(err: AddressParseError) -> Self { Self::AddressParseError(err.to_string()) }
+}
+
+impl From<serialization::Error> for CheckPaymentSentError {
+    fn from(err: serialization::Error) -> Self { Self::DeserializationErr(err.to_string()) }
+}
+
+impl From<UnexpectedDerivationMethod> for CheckPaymentSentError {
+    fn from(err: UnexpectedDerivationMethod) -> Self { Self::UnexpectedDerivationMethod(err) }
+}
+
+#[derive(Debug, Display, PartialEq)]
+pub enum ExtractSecretError {
+    DeserializationErr(String),
+    DecodingError(String),
+    ExtractionFailed(String),
+    CouldNotObtainSecret(String),
+    #[display(fmt = "Invalid arguments in 'receiverSpend' call: {:?}", _0)]
+    InvalidArguments(Vec<ethabi::Token>),
+    #[display(fmt = "Expected secret to be fixed bytes, decoded function data is {:?}", _0)]
+    ExpectedFixedBytes(Vec<ethabi::Token>),
+}
+
+impl From<ethabi::Error> for ExtractSecretError {
+    fn from(err: ethabi::Error) -> Self { Self::DecodingError(err.to_string()) }
+}
+
+impl From<rlp::DecoderError> for ExtractSecretError {
+    fn from(err: rlp::DecoderError) -> Self { Self::DecodingError(err.to_string()) }
+}
+
+impl From<serialization::Error> for ExtractSecretError {
+    fn from(err: serialization::Error) -> Self { Self::DeserializationErr(err.to_string()) }
+}
+
+#[derive(Debug, Display, PartialEq)]
+pub enum MyAddressError {
```

Comment:
`MyAddressError` needs to be refactored significantly.
First of all, please remove unused error variants. `MethodNotSupported`, `TxReadError`, `UtxoRpcError` are not used at all.
Secondly, tI see only two error types that the `MarketCoinOps::my_address` method caller may care about:
```rust
pub enum MyAddressError {
  UnexpectedDerivationMethod(UnexpectedDerivationMethod),
  Internal(String),
}
```
If `my_address` failed due to an unexpected derivation method, then it seems that the user called an RPC on a coin that is activated with an HD derivation method. Otherwise we expect the method to succeed and return `Internal` error instead.